pkg/ddl/partition.go

```diff
@@ -3279,6 +3285,8 @@ func (w *worker) onReorganizePartition(jobCtx *jobContext, job *model.Job) (ver
 				newIndex := index.Clone()
 				newIndex.State = model.StateDeleteOnly
 				newIndex.ID = AllocateIndexID(tblInfo)
+				tblInfo.Partition.DDLChangedIndex[index.ID] = false
+				tblInfo.Partition.DDLChangedIndex[newIndex.ID] = true
 				newIndex.Global = true
 				tblInfo.Indices = append(tblInfo.Indices, newIndex)
 			} else {
```

Comment:
I forget remove L3293 in https://github.com/pingcap/tidb/pull/56868, maybe we could remove it in this PR.

Reply:
Done.